paperspace/cli.py

```diff
@@ -0,0 +1,208 @@
+import functools
+import json
+
+import click
+
+from paperspace import commands, constants
+
+
+class ChoiceType(click.Choice):
+    """Takes a string-keyed map and converts cli-provided parameter to corresponding value"""
+
+    def __init__(self, type_map, case_sensitive=True):
+        super(ChoiceType, self).__init__(tuple(type_map.keys()), case_sensitive=case_sensitive)
+        self.type_map = type_map
+
+    def convert(self, value, param, ctx):
+        value = super(ChoiceType, self).convert(value, param, ctx)
+        return self.type_map[value]
+
+
+EXPERIMENT_TYPES_MAP = {
+    "GRPC": constants.EXPERIMENT_TYPE_GRPC_MULTI_NODE_ID,
+    "MPI": constants.EXPERIMENT_TYPE_MPI_MULTI_NODE_ID,
+}
+
+
+def json_string(val):
+    """Wraps json.loads so the cli help shows proper option's type name instead of 'LOADS'"""
+    return json.loads(val)
+
+
+def del_if_value_is_none(d):
+    """Remove all elements with value == None"""
+    for key, val in list(d.items()):
+        if val is None:
+            del d[key]
+
```

Comment:
i'm not really sure naming parameter just `d` is a good idea...
also nit: check out more pythonic way:
`return  {k: v for k, v in d.items() if v is not None}

Reply:
name changed do dict_

this won't modify the existing dict